Velaptor/Scene/SceneManager.cs

```diff
@@ -0,0 +1,228 @@
+// <copyright file="SceneManager.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+namespace Velaptor.Scene;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Velaptor;
+
+/// <summary>
+/// Manages scenes by loading and unloading content, updating, and rendering scenes.
+/// </summary>
+public sealed class SceneManager : ISceneManager
+{
+    private readonly List<(IScene? scene, bool isActive)> scenes = new ();
+    private bool isDisposed;
+    private bool isLoaded;
+
+    /// <inheritdoc/>
+    public IScene? CurrentScene => this.scenes.FirstOrDefault(s => s.isActive).scene;
+
+    /// <inheritdoc/>
+    public IReadOnlyCollection<Guid> InActiveScenes =>
+        this.scenes.Where(s => s.scene is not null && s.isActive is false)
+            .Select(s => s.scene?.Id ?? Guid.Empty).ToArray().AsReadOnly();
+
+    /// <inheritdoc/>
+    /// <remarks>
+    ///     If no scenes exist in the manager, the scene will be active even if the <paramref name="setToActive"/>
+    ///     parameter is set to <c>false</c>.
+    ///     <br/>
+    ///     This is because there has to be one active scene at all times.
+    /// </remarks>
+    /// <exception cref="Exception">
+    ///     Thrown if a scene with with the given <paramref name="scene"/>'s ID already exists.
+    /// </exception>
+    public void AddScene(IScene scene, bool setToActive = false)
+    {
+        if (SceneExists(scene.Id))
+        {
+            var midSection = string.IsNullOrEmpty(scene.Name)
+                ? $"with the ID '{scene.Id}'"
+                : $"'{scene.Name}' with the ID '{scene.Id}'";
+            var exceptionMsg = $"The scene {midSection} already exists.";
+
+            // TODO: Create custom exception for this
```

Comment:
Do you still need this TODO?
```suggestion
            // TODO: Create custom exception for this
```

Reply:
New exception created and test updated